Sources/Sentry/SentryHub.m

```diff
@@ -511,16 +534,38 @@ - (SentryId *)captureException:(NSException *)exception
 
 - (SentryId *)captureException:(NSException *)exception withScope:(SentryScope *)scope
 {
+    SentryId * (^captureClientBlock)(SentryClientInternal *) = ^SentryId *(
```

Comment:
`m`: In previous PR feedback you mentioned that we should refrain from using NSBlock due to it's instability and memory unsafety. We should consider using private methods instead.